### PR TITLE
ClosureSourceLocator improvements

### PR DESCRIPTION
**Pull request by kukulich**
https://github.com/Roave/BetterReflection/issues/228 and https://github.com/Roave/BetterReflection/issues/152 should be solved.

### CODE REVIEW

File: src/SourceLocator/Type/ClosureSourceLocator.php [hunk 1]
```diff
@@ -4,100 +4,159 @@
...
+                if (isset($closureNodesDataOnSameLine[1])) {
+                    throw TwoClosuresOnSameLine::create($this->fileName, $this->startLine);
+                }
+
+                return $closureNodesDataOnSameLine[0];
+            }
+        };
+
+        $fileContents = \file_get_contents($fileName);
+        $ast          = (new ParserFactory())->create(ParserFactory::PREFER_PHP7)->parse($fileContents);
```
Comment (Ocramius) [inline src/SourceLocator/Type/ClosureSourceLocator.php]: Note to self: one more parser to inject from an outer scope

Comment (Ocramius) [inline src/SourceLocator/Type/ClosureSourceLocator.php]: Move this call to the constructor, as I may need to move the parser to an injected dependency shortly

File: src/SourceLocator/Type/ClosureSourceLocator.php [hunk 2]
```diff
@@ -4,100 +4,159 @@
...
+
+            /**
+             * @var int
+             */
+            private $startLine;
+
+            /**
+             * @var Node\Expr\Closure[]
+             */
+            private $closureNodesData = [];
```
Comment (Ocramius) [inline src/SourceLocator/Type/ClosureSourceLocator.php]: Just `closureNodes`

Comment (kukulich) [inline src/SourceLocator/Type/ClosureSourceLocator.php]: Done.

File: src/SourceLocator/Type/ClosureSourceLocator.php (hunk 1 above)
Comment (kukulich) [inline src/SourceLocator/Type/ClosureSourceLocator.php]: Done.

**Comment by asgrim** [comment]
Thanks @kukulich! I've been wanting to get this done for ages but never found the time :grinning:

**Pull request merged**